skyportal/handlers/api/tns.py

```diff
@@ -0,0 +1,592 @@
+            return self.error(
+                f'Error parsing posted tnsrobot: "{e.normalized_messages()}"'
+            )
+
+        group = Group.get_if_accessible_by(tnsrobot.group_id, self.current_user)
+        if group is None:
+            return self.error(f'No group with specified ID: {tnsrobot.group_id}')
+
+        DBSession().add(tnsrobot)
+        self.verify_and_commit()
+        return self.success(data={"id": tnsrobot.id})
+
+    @permissions(['Manage tnsrobots'])
+    def delete(self, tnsrobot_id):
+        """
+        ---
+        description: Delete TNS robot.
+        tags:
+          - tnsrobots
+        parameters:
+          - in: path
+            name: tnsrobot_id
+            required: true
+            schema:
+              type: string
+        responses:
+          200:
+            content:
+              application/json:
+                schema: Success
+        """
+        tnsrobot = TNSRobot.get_if_accessible_by(
+            int(tnsrobot_id), self.current_user, mode='delete'
+        )
+        DBSession().delete(tnsrobot)
+        self.verify_and_commit()
+        return self.success()
+
+
+class ObjTNSHandler(BaseHandler):
+    @auth_or_token
+    def post(self, obj_id):
+        """
+        ---
+        description: Post an Obj to TNS
+        tags:
+          - objs
+        parameters:
+          - in: path
+            name: obj_id
+            required: true
+            schema:
+              type: string
+        responses:
+          200:
+            content:
+              application/json:
+                schema: Success
+          400:
+            content:
+              application/json:
+                schema: Error
+        """
+
+        obj = Obj.get_if_accessible_by(
+            obj_id,
+            self.current_user,
+            mode='read',
+            raise_if_none=True,
+        )
+
+        photometry = (
+            Photometry.query_records_accessible_by(self.current_user)
+            .filter(Photometry.obj_id == obj_id)
+            .all()
+        )
+        photometry = [serialize(phot, 'ab', 'mag') for phot in photometry]
+
+        data = self.get_json()
+        tnsrobotID = data.get('tnsrobotID')
+        reporters = data.get('reporters', '')
+
+        if tnsrobotID is None:
+            return self.error('tnsrobotID is required')
+
+        tnsrobot = DBSession().query(TNSRobot).filter(TNSRobot.id == tnsrobotID).one()
+
+        altdata = tnsrobot.altdata
+        if not altdata:
+            raise ValueError('Missing TNS information.')
+
+        tns_headers = {
+            'User-Agent': 'tns_marker{"tns_id":'
+            + str(tnsrobot.bot_id)
+            + ', "type":"bot", "name":"'
+            + tnsrobot.bot_name
+            + '"}'
+        }
+
+        time_first, mag_first, magerr_first, filt_first, instrument_first = (
+            None,
+            None,
+            None,
+            None,
+            None,
+        )
```

Comment:
```suggestion
        time_first = mag_first = magerr_first = filt_first = instrument_first = None
```
Although there's a chance one of the lint/black formatters will not like this...